ccl/changefeedccl: TestParquetRows failed

ccl/changefeedccl.TestParquetRows [failed](https://tanzanite.cluster.engflow.com/invocation/be50d37f-c771-45ca-a89d-a7d3b521b6a5) on master @ [8315a4bc997fb8b8d679079e14d6d7ca94d53bc6](https://github.com/cockroachdb/cockroach/commits/8315a4bc997fb8b8d679079e14d6d7ca94d53bc6):


```
=== RUN   TestParquetRows
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestParquetRows4277821759
    test_log_scope.go:81: use -show-logs to present logs inline
    conditional_wrap.go:190: 
        pkg/ccl/changefeedccl/parquet_test.go:235: (makeRangefeedReaderAndDecoder)
        	NOTICE: .ExecutorConfig() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().ExecutorConfig() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    parquet_test.go:230: -- test log scope end --
test logs left over in: outputs.zip/logTestParquetRows4277821759
--- FAIL: TestParquetRows (0.96s)
=== RUN   TestParquetRows/mixed
    testutils.go:427: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/parquet/testutils.go:427
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/parquet_test.go:225
        	Error:      	Not equal: 
        	            	expected: (*tree.DString)(0xc0016ac260)
        	            	actual  : (*tree.DString)(0xc00bc6bab0)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(*tree.DString)((len=7) "changed")
        	            	+(*tree.DString)((len=2) "d1")
        	            	 
        	Test:       	TestParquetRows/mixed
    parquet_test.go:157: leaving /var/lib/engflow/worker/work/3/exec/_tmp/18f3db96b6c041b0f0bfe1c2789226f6/TestParquetRows2197263694 for inspection
    --- FAIL: TestParquetRows/mixed (0.16s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=14</code>
, <code>shard=14</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParquetRows.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32719

--- a/pkg/ccl/changefeedccl/parquet_test.go
+++ b/pkg/ccl/changefeedccl/parquet_test.go
@@ -214,13 +214,14 @@ func TestParquetRows(t *testing.T) {
 			// NB: Rangefeeds have per-key ordering, so the rows in the parquet
 			// file may not match the order we insert them. To accommodate for
 			// this, sort the expected and actual datums by the primary key.
-			slices.SortFunc(datums, func(a []tree.Datum, b []tree.Datum) bool {
+			slices.SortStableFunc(datums, func(a []tree.Datum, b []tree.Datum) bool {
 				return a[0].Compare(&eval.Context{}, b[0]) == -1
 			})
-			slices.SortFunc(readDatums, func(a []tree.Datum, b []tree.Datum) bool {
+			slices.SortStableFunc(readDatums, func(a []tree.Datum, b []tree.Datum) bool {
 				return a[0].Compare(&eval.Context{}, b[0]) == -1
 			})
 			for r := 0; r < numRows; r++ {
+				t.Logf("comparing row expected: %s to actual: %s\n", datums[r], readDatums[r])
 				for c := 0; c < numCols; c++ {
 					parquet.ValidateDatum(t, datums[r][c], readDatums[r][c])
 				}

